framework/wcf/Extensibility/Transport/AzureQueueStorageClient/AzureQueueStorageClient/SampleProfileAzureQueueStorageBinding.cs

```diff
@@ -0,0 +1,434 @@
+﻿//----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//----------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Configuration;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+
+namespace Microsoft.Samples.AzureQueueStorage
+{
+
+    /// <summary>
+    /// Binding for Udp. This is our "sample profile" for Udp, which uses Text+Soap 1.2 
+    /// and allows for variation in Reliability capabilities. If ReliableSessionEnabled is set
+    /// then we will layer RM+CompositeDuplex on top of Udp. Otherwise we will just
+    /// have Udp on our stack.  
+    /// </summary>
+    public class SampleProfileAzureQueueStorageBinding : Binding
+    {
+        bool reliableSessionEnabled;
+
+        // private BindingElements
+        //CompositeDuplexBindingElement compositeDuplex;
+        ReliableSessionBindingElement session;
+        AzureQueueStorageTransportBindingElement transport;
+        MessageEncodingBindingElement encoding;
+
+        public SampleProfileAzureQueueStorageBinding()
+        {
+            Initialize();
+        }
+
+        public SampleProfileAzureQueueStorageBinding(bool reliableSessionEnabled) : this()
+        {
+            this.ReliableSessionEnabled = reliableSessionEnabled;
+        }
+
+        /*public SampleProfileUdpBinding(string configurationName) : this()
+        {
+            ApplyConfiguration(configurationName);
+        }*/
+
+        public bool OrderedSession
+        {
+            get { return session.Ordered; }
+            set { session.Ordered = value; }
+        }
+
+        public bool ReliableSessionEnabled
+        {
+            get { return reliableSessionEnabled; }
+            set { reliableSessionEnabled = value; }
+        }
+
+        public override string Scheme { get { return "soap.udp"; } }
+
+        public TimeSpan SessionInactivityTimeout
+        {
+            get { return this.session.InactivityTimeout; }
+            set { this.session.InactivityTimeout = value; }
+        }
+
+        public EnvelopeVersion SoapVersion
+        {
+            get { return EnvelopeVersion.Soap12; }
+        }
+
+        /*public Uri ClientBaseAddress
+        {
+            get { return this.compositeDuplex.ClientBaseAddress; }
+            set { this.compositeDuplex.ClientBaseAddress = value; }
+        }*/
+
+        /// <summary>
+        /// Create the set of binding elements that make up this binding. 
+        /// NOTE: order of binding elements is important.
+        /// </summary>
+        /// <returns></returns>
+        public override BindingElementCollection CreateBindingElements()
+        {   
+            BindingElementCollection bindingElements = new BindingElementCollection();
+
+            if (ReliableSessionEnabled)
+            {
```

Comment:
Remove this if block. ReliableSessions only makes sense when you have a reply path that the client can listen to. We don't support compositeDuplex which is needed to do this.

Reply:
removed as part of above cleanups